src/jit/emit.cpp

```diff
@@ -4975,12 +5005,16 @@ unsigned emitter::emitEndCodeGen(Compiler* comp,
 
     unsigned actualCodeSize = emitCurCodeOffs(cp);
 
-    /* Fill in eventual unused space */
+    // Fill in eventual unused space, but do not report this space as used.
+    // If you add this padding during the emitIGlist loop, then it will
+    // emit offsets after the loop with wrong value (for example for GC ref variables).
     while (emitCurCodeOffs(cp) < emitTotalCodeSize)
     {
         *cp++ = DEFAULT_CODE_BUFFER_INIT;
     }
 
+    emitTotalCodeSize = actualCodeSize;
```

Comment:
Putting this here makes the `totAllocdSize` addition below incorrect. (Although maybe it's already incorrect, if it doesn't match what was actually allocated when we called allocMem)